src/test/java/teammates/ui/webapi/GetFeedbackSessionActionTest.java

```diff
@@ -91,6 +96,821 @@ protected void testExecute() {
+        assertEquals(FeedbackSessionPublishStatus.NOT_PUBLISHED, response.getPublishStatus());
+
+        assertEquals(relativeFeedbackSession.isClosingEmailEnabled(), response.getIsClosingEmailEnabled());
+        assertEquals(relativeFeedbackSession.isPublishedEmailEnabled(), response.getIsPublishedEmailEnabled());
+
+        assertEquals(relativeFeedbackSession.getCreatedTime().toEpochMilli(), response.getCreatedAtTimestamp());
+        assertNull(response.getDeletedAtTimestamp());
+
+        assertEqualDeadlines(relativeFeedbackSession.getStudentDeadlines(), response.getStudentDeadlines(), timeZone);
+        assertEqualDeadlines(relativeFeedbackSession.getInstructorDeadlines(), response.getInstructorDeadlines(),
+                timeZone);
+
+        ______TS("get full detail; no extensions; after end time but within grace period");
+
+        relativeStudentDeadlines = new HashMap<>();
+        relativeInstructorDeadlines = new HashMap<>();
+        relativeFeedbackSession = generateRelativeFeedbackSessionInTypicalCourse1(-60, relativeStudentDeadlines,
+                relativeInstructorDeadlines);
+        typicalBundle.feedbackSessions.put("relativeFeedbackSession", relativeFeedbackSession);
+        removeAndRestoreDataBundle(typicalBundle);
+        a = getAction(params);
+        r = getJsonResult(a);
+        response = (FeedbackSessionData) r.getOutput();
+
+        assertEquals(FeedbackSessionSubmissionStatus.GRACE_PERIOD, response.getSubmissionStatus());
+
+        assertEqualDeadlines(relativeFeedbackSession.getStudentDeadlines(), response.getStudentDeadlines(), timeZone);
+        assertEqualDeadlines(relativeFeedbackSession.getInstructorDeadlines(), response.getInstructorDeadlines(),
+                timeZone);
+
+        ______TS("get full detail; no extensions; after end time and beyond grace period");
+
+        relativeStudentDeadlines = new HashMap<>();
+        relativeInstructorDeadlines = new HashMap<>();
+        relativeFeedbackSession = generateRelativeFeedbackSessionInTypicalCourse1(-60 * 60 * 24, relativeStudentDeadlines,
+                relativeInstructorDeadlines);
+        typicalBundle.feedbackSessions.put("relativeFeedbackSession", relativeFeedbackSession);
+        removeAndRestoreDataBundle(typicalBundle);
+        a = getAction(params);
+        r = getJsonResult(a);
+        response = (FeedbackSessionData) r.getOutput();
+
+        assertEquals(FeedbackSessionSubmissionStatus.CLOSED, response.getSubmissionStatus());
+
+        assertEqualDeadlines(relativeFeedbackSession.getStudentDeadlines(), response.getStudentDeadlines(), timeZone);
+        assertEqualDeadlines(relativeFeedbackSession.getInstructorDeadlines(), response.getInstructorDeadlines(),
+                timeZone);
+
+        ______TS("get full detail; some extensions; before end time");
+
+        relativeStudentDeadlines.put("student1InCourse1@gmail.tmt", 60L * 60 * 21);
```

Comment:
Do we need to define relative student deadlines? Seems a little unnecessarily complicated to me, especially for a test case. Any issue with just adding the Instant directly? Something along the lines of:
```
Instant now = Instant.now();
studentDeadlines.put("...", now.plusSeconds(...));
```

Reply:
I think I tried something similar but I changed it because I needed to create the `relativeFeedbackSession` every time to call `removeAndRestoreDataBundle` later on. However, with the feedback session methods from logic, I probably would not need to do that anymore.